python/lsst/summit/utils/guiders/seeing.py

```diff
@@ -0,0 +1,517 @@
+            starsFiltered, dummyNmissingStampPlaceholder
+        )
+        self.metrics: pd.DataFrame = metricsBuilder.buildMetrics(self.expid)
+        starsDetrended: pd.DataFrame = metricsBuilder.detrendStars()
+
+        # Build star list per detector with a flux threshold.
+        self.starList: dict[str, pd.DataFrame] = groupStarsPerDetector(starsDetrended, fluxCut)
+        self.detectors: list[str] = list(self.starList.keys())
+
+    def measureVar(self) -> list[float]:
+        """
+        Compute per-detector variances of corrected centroid motions.
+
+        Requires that detrending has already been applied. Variances are
+        scaled to units of (pix × 100)^2, i.e., (centipix)^2.
+
+        Returns
+        -------
+        out : `list[float]`
+            Flattened list of per-detector variances:
+            ``[vx_det1, vy_det1, vx_det2, vy_det2, ...]``. If a detector
+            was excluded or lacks corrected coordinates, entries are
+            ``NaN``.
+        """
+        out: list[float] = []
+        for _, df in self.starList.items():
+            meanFlux = float(np.nanmean(pd.to_numeric(df["flux"], errors="coerce")))
+            hasCorr = "dxfp_corr" in df and "dyfp_corr" in df
+            if meanFlux > self.fluxCut and hasCorr:
+                vx = float(np.nanvar(pd.to_numeric(df["dxfp_corr"], errors="coerce"))) * (100.0**2)
+                vy = float(np.nanvar(pd.to_numeric(df["dyfp_corr"], errors="coerce"))) * (100.0**2)
+                out.extend([vx, vy])
+            else:
+                out.extend([np.nan, np.nan])
+        return out
+
+    @staticmethod
+    def makePairs(kind: str) -> list[list[str]]:
+        """
+        Produce preset detector pairs.
+
+        Parameters
+        ----------
+        kind : `str`
+            One of ``{'adjacent','cross','diagonal'}``.
+
+        Returns
+        -------
+        pairs : `list[list[str]]`
+            List of 2-element lists ``[det1, det2]`` describing pairs.
+
+        Raises
+        ------
+        KeyError
+            If ``kind`` is not a recognized preset.
+        """
+        presets: dict[str, list[list[str]]] = {
+            "adjacent": [
+                ["R00_SG0", "R00_SG1"],
+                ["R04_SG0", "R04_SG1"],
+                ["R40_SG0", "R40_SG1"],
+                ["R44_SG0", "R44_SG1"],
+            ],
+            "cross": [
+                ["R00_SG0", "R40_SG1"],
+                ["R04_SG1", "R44_SG0"],
+                ["R00_SG1", "R04_SG0"],
+                ["R40_SG0", "R44_SG1"],
+            ],
+            "diagonal": [
+                ["R00_SG0", "R44_SG1"],
+                ["R00_SG1", "R44_SG0"],
+                ["R40_SG0", "R04_SG1"],
+                ["R40_SG1", "R04_SG0"],
+            ],
+        }
+        return presets[kind]
+
+    def measureCorrPairs(
+        self, kind: str
+    ) -> tuple[NDArray[np.float64], NDArray[np.float64], NDArray[np.float64]]:
```

Comment:
This might be OK for now, but I tend to prefer dataclasses when you start returning list of arrays. Keep it as-is for now, but just something to consider for the future. It also saves you from if you change a return order, because when you use the dataclass that's returned, you'll be accessing things by name, rather than order, so it's more robust to changes in the future.